CMake/kwiver-setup-python.cmake

```diff
@@ -204,6 +204,121 @@ mark_as_advanced(PYTHON_ABIFLAGS)
+    # require conda specific command
+    set(CONDA 1)
+    set(VENV_DIR "testing_venv")
+    set(CREATE_VENV "conda" "create" "--name" ${VENV_DIR} "python")
+  else()
+    set(VENV_DIR "${KWIVER_BINARY_DIR}/testing_venv")
+    set(CREATE_VENV "python3" "-m" "venv" ${VENV_DIR} )
+  endif()
+  message(STATUS "Creating virtualenv @${VENV_DIR} for testing...")
+  execute_process (
+                    COMMAND ${CREATE_VENV}
+                    RESULT_VARIABLE create_venv_result
+                    ERROR_VARIABLE create_venv_error
+                  )
+
+  if(create_venv_result AND NOT create_venv_result EQUAL 0)
+      # could not create venv, report to that effect, Nose may still be found and tests may still be run
+      # but dependencies (including nose) to be met by a pip install are not garunteed
+      message (WARNING "Virtualenv creation failed, Python tests may not be run or may fail unexpectedly.\
+                        Python Error: ${create_venv_error}")
+  else()
+    set(VENV_CREATED 1)
+  endif()
+  message(STATUS "Virtualenv creation exited with status: ${create_venv_result}")
+  unset(Python3_FOUND)
+  if(VENV_CREATED)
+    if( APPLE )
+      set(CMAKE_FIND_FRAMEWORK "NEVER")
+    elseif( WIN32 )
+      set(Python3_FIND_REGISTRY "NEVER")
+    endif()
+    include("${KWIVER_SOURCE_DIR}/CMake/utils/kwiver-config-venv.cmake")
+    ACTIVATE_VENV(${VENV_DIR})
+    set(ENV{VIRTUAL_ENV} ${VENV_DIR})
+    set(Python3_FIND_VIRTUALENV ONLY)
+    unset( Python3_EXECUTABLE)
+    find_package(Python3 COMPONENTS Interpreter Development)
+    if( NOT Python3_FOUND)
+      message(WARNING
+              "Could not find virtualenv Python interp.\
+              Python tests may be run without garuntee of dependencies")
+    else()
+      # conda comes with a pip install so this should be flavor agnostic
+      set(PIP_UPGRADE_COMMAND "-m" "pip" "-q" "install" "--upgrade" "pip")
+      set(PIP_COMMAND "${Python3_EXECUTABLE}"
+                      "-m"
+                      "pip"
+                      "-q"
+                      "install"
+                      "-r"
+                      "${KWIVER_SOURCE_DIR}/python/requirements.txt"
+                      "||"
+                      "${CMAKE_COMMAND}" "-E" "echo"
+                      "Pip install failed, consult build output. Python dependencies may not be met"
+                      )
+      message(STATUS "Test dependencies will be installed via pip to: ${VENV_DIR}")
+
+    endif()
+    DEACTIVATE_VENV()
+  endif()
+
+  ###
+  # Pybind11 Bindings Test Runner - nosetests
+  # find virtualenv install of nosetests executable, search for version associated with kwiver
+  # first, default to v 3.4, the most recent version provided by pip install
+  # alternatively users can install the version of nose specific
+  # to the version of python they're building the kwiver-python against
+  #
+  #
+
+  if(Python3_FOUND)
+    #TODO: if this point is executed, venv is created. Need to add sourcing
```

Comment:
Is this still TODO?

Reply:
No, and I moved away from that implementation anyway, so it's both out of date and misleading, I'll remove it.